### PR TITLE
Gogo golang/migration

### PR DESCRIPTION
**Pull request by gandhikrishna**
- [ ] Description and context for reviewers: one partner, one stranger
- [ ] Docs (package doc)
- [ ] Entry in CHANGELOG.md

### CODE REVIEW

**Comment by Dogild** [comment]
PLease rebase with dev

File: encoding/protobuf/error.go [hunk 1]
```diff
@@ -25,12 +25,12 @@ import (
...
 	"strings"
 
-	"github.com/gogo/googleapis/google/rpc"
-	"github.com/gogo/protobuf/proto"
-	"github.com/gogo/status"
+	"github.com/golang/protobuf/proto"
 	"go.uber.org/yarpc/api/transport"
 	"go.uber.org/yarpc/internal/grpcerrorcodes"
 	"go.uber.org/yarpc/yarpcerrors"
+	spb "google.golang.org/genproto/googleapis/rpc/status"
```
Comment (Dogild) [inline encoding/protobuf/error.go:32]: nit: let's rename this statuspb, even if spb is kind of the same it will help for readability of the code

File: encoding/protobuf/error_integration_test.go [hunk 2]
```diff
@@ -40,14 +38,16 @@ import (
 	"go.uber.org/yarpc/encoding/raw"
 	"go.uber.org/yarpc/transport/grpc"
 	"go.uber.org/yarpc/yarpcerrors"
+	rpcStatus "google.golang.org/genproto/googleapis/rpc/status"
```
Comment (Dogild) [inline encoding/protobuf/error_integration_test.go:41]: nit: let's rename this statuspb, even if spb is kind of the same it will help for readability of the code. Also make sure that it is the same name in yarpc everywhere

File: encoding/protobuf/error_integration_test.go [hunk 3]
```diff
@@ -116,11 +116,13 @@ func TestProtoGrpcServerErrorDetails(t *testing.T) {
...
-		&types.StringValue{Value: "string value"},
-		&types.Int32Value{Value: 100},
+		&wrappers.StringValue{Value: "string value"},
+		&wrappers.Int32Value{Value: 100},
 	}
 	actualDetails := protobuf.GetErrorDetails(err)
-	assert.Equal(t, expectedDetails, actualDetails, "unexpected error details")
+	if !proto.Equal(expectedDetails[0].(proto.Message), actualDetails[0].(proto.Message)) {
+		t.Errorf("unexpected error details")
+	}
```
Comment (Dogild) [inline encoding/protobuf/error_integration_test.go:125]: This should be `assert.True(t, proto.Equal(expectedDetails[0].(proto.Message), actualDetails[0].(proto.Message))`

Why did you need to change this test?

File: encoding/protobuf/error_integration_test.go [hunk 4]
```diff
@@ -185,15 +187,17 @@ func TestProtoGrpcStreamServerErrorDetails(t *testing.T) {
 	assert.Equal(t, errorMsg, st.Message(), "unexpected error message")
 
 	actualDetails := protobuf.GetErrorDetails(err)
-	assert.Equal(t, expectedDetails, actualDetails, "unexpected error details")
+	if !proto.Equal(expectedDetails[0].(proto.Message), actualDetails[0].(proto.Message)) {
+		t.Errorf("unexpected error details")
+	}
```
Comment (Dogild) [inline encoding/protobuf/error_integration_test.go:192]: This should be `assert.True(t, proto.Equal(expectedDetails[0].(proto.Message), actualDetails[0].(proto.Message))`

Why did you need to change this test?

File: encoding/protobuf/error_test.go [hunk 5]
```diff
@@ -184,7 +185,9 @@ func TestPbErrorToYARPCError(t *testing.T) {
 			assert.Equal(t, tt.message, status.Message(), "unexpected grpc status message")
 			assert.Len(t, status.Details(), len(tt.details), "unexpected details length")
 			for i, detail := range tt.details {
-				assert.Equal(t, detail, status.Details()[i])
+				if !proto.Equal(detail, status.Details()[i].(proto.Message)) {
+					t.Errorf("non comparable messages")
+				}
```
Comment (Dogild) [inline encoding/protobuf/error_test.go:190]: This should be `assert.True(t, proto.Equal(detail, status.Details()[i].(proto.Message)))`

File: encoding/protobuf/stream_test.go [hunk 6]
```diff
@@ -75,7 +75,7 @@ func TestUnary(t *testing.T) {
 			sent := &testpb.TestMessage{Value: "echo"}
 			received, err := client.Unary(ctx, sent)
 			require.NoError(t, err)
-			assert.Equal(t, sent, received)
+			assert.Equal(t, sent.Value, received.Value)
```
Comment (Dogild) [inline encoding/protobuf/stream_test.go:78]: Why did you need to change this test?

File: internal/examples/go.mod [hunk 7]
```diff
@@ -12,6 +11,7 @@ require (
 	go.uber.org/yarpc v1.42.1
 	go.uber.org/zap v1.13.0
 	google.golang.org/grpc v1.27.1
+	google.golang.org/protobuf v1.25.0
```
Comment (Dogild) [inline internal/examples/go.mod:14]: Why do we have this reference of golang/protov2? the goal was only to move to `github.com/golang/protobuf 1.3.5` no?

File: encoding/protobuf/error_integration_test.go (hunk 3 above)
Comment (gandhikrishna) [reply encoding/protobuf/error_integration_test.go:125]: Since we are running in mix mode `gogo <-> google/protobuf` structs(err details in this case) are not equal. This is due to extra added semi-private fields in the structs which are used by the library internally for various usecases.  And hence the `assert.equal` won't work. So using `proto.equal` is more preferable here.

File: internal/examples/go.mod (hunk 7 above)
Comment (gandhikrishna) [reply internal/examples/go.mod:14]: I don't see any single usage of `google.golang.org/protobuf` in the yarpc code base. Let me check if this is due to indirect dependency.
